x/farming/spec/02_state.md

```diff
@@ -147,8 +147,8 @@ type QueuedStaking struct {
 }
 ```
 
-- QueuedStaking: `0x23 | StakingCoinDenomLen (1 byte) | StakingCoinDenom | FarmerAddr -> ProtocolBuffer(QueuedStaking)`
-- QueuedStakingIndex: `0x24 | FarmerAddrLen (1 byte) | FarmerAddr | StakingCoinDenom -> nil`
+- QueuedStaking: `0x23 | EndTimeLen (1 byte) | sdk.FormatTimeBytes(EndTime) | StakingCoinDenomLen (1 byte) | StakingCoinDenom | FarmerAddr -> ProtocolBuffer(QueuedStaking)`
+- QueuedStakingIndex: `0x24 | FarmerAddrLen (1 byte) | FarmerAddr | StakingCoinDenomLen (1 byte) | StakingCoinDenom | sdk.FormatTimeBytes(EndTime) -> nil`
```

Comment:
Don't we need `EndTimeLen (1 byte)` on `QueuedStakingIndex` also?

Reply:
As same as other keys, I just omitted the length prefix for the last element in the key.
Please see that `QueuedStaking` has no `FarmerAddrLen (1 byte)` prefix for `FarmerAddr` too.